Actions/IncrementVersionNumber/IncrementVersionNumber.psm1

```diff
@@ -0,0 +1,139 @@
+    $settingFileContent.$settingName = $newValue.ToString()
+    $settingFileContent | Set-JsonContentLF -Path $settingsFilePath
+
+    return @{
+        name = $settingFileContent.name
+        version = $settingFileContent.version
+        publisher = $settingFileContent.publisher
+    }
+}
+
+<#
+    .Synopsis
+        Changes the version number of a project.
+    .Description
+        Changes the version number of a project.
+        The version number is changed in the project settings file (value for 'repoVersion') and in the app.json files of all apps in the project, as well as all references to the apps in the dependencies of the app.json files.
+    .Parameter baseFolder
+        Base folder of the repository.
+    .Parameter project
+        Name of the project (relative to the base folder).
+    .Parameter newVersion
+        New version number. If the version number starts with a +, the new version number will be added to the old version number. Else the new version number will replace the old version number.
+#>
+function Set-ProjectVersion($projectPath, $projectSettings, $newVersion, $ALGOsettingsFile) {
+    # Set repoVersion in project settings (if it exists)
+    $projectSettingsPath = Join-Path $projectPath $ALGoSettingsFile
+    Set-VersionSettingInFile -settingsFilePath $projectSettingsPath -settingName 'repoVersion' -newValue $newVersion | Out-Null
+
+    # Check if the project uses repoVersion versioning strategy
+    $useRepoVersion = (($projectSettings.PSObject.Properties.Name -eq "versioningStrategy") -and (($projectSettings.versioningStrategy -band 16) -eq 16))
+    if ($useRepoVersion) {
+        $newVersion = $projectSettings.repoVersion
+    }
+
+    $allAppFolders = @($projectSettings.appFolders) + @($projectSettings.testFolders) + @($projectSettings.bcptTestFolders)
+    # Set version in app.json files
+    $appInfos = $allAppFolders | ForEach-Object {
+        $folder = Join-Path $projectPath $_
+        $appJsonFile = Join-Path $folder "app.json"
+
+        $appInfo = Set-VersionSettingInFile -settingsFilePath $appJsonFile -settingName 'version' -newValue $newVersion
+        return $appInfo
+    }
+
+    # Set the version in the dependencies in app.json files
+    $allAppFolders | ForEach-Object {
+        $folder = Join-Path $projectPath $_
+        $appJsonFile = Join-Path $folder "app.json"
+
+        $appJsonContent = Get-Content $appJsonFile -Raw -Encoding UTF8 | ConvertFrom-Json
+        $dependencies = $appJsonContent.dependencies
+        if ($null -ne $dependencies) {
+            $dependencies | ForEach-Object {
+                $dependency = $_
+                # Find the version of the dependency in the appInfos. If it's found, it means the dependency is part of the project and the version should be set.
+                $dependencyAppInfo = $appInfos | Where-Object { $_.name -eq $dependency.name -and $_.publisher -eq $dependency.publisher }
```

Comment:
This wouldn't work if the dependency app is from another AL-Go project.
`$appInfos` contains the apps from the current AL-Go project.

Reply:
Yeah - I guess we should resolve all appfolders for all projects and then call into this function with multi-project wide appfolders instead of a single project.

Reply:
Implemented. Testing...

Reply:
Test run: https://github.com/mazhelez/BCApps/actions/runs/7833162162